lib/grape/jsonapi/entity/resource.rb

```diff
@@ -28,6 +28,30 @@ def included_exposure
               end
             end
           end
+
+          private
+
+          def avoid_recursive_relation_check(nested_class)
+            return false if nested_class.root_exposures.empty?
+            nesting_exposures = nested_class.root_exposures.select do |exposure|
+              exposure.class == Grape::Entity::Exposure::NestingExposure
+            end
+            return false if nesting_exposures.empty?
+            result = nesting_exposures.detect do |exposure|
+              recursive_exposure?(exposure)
+            end
+            result
+          end
+
+          def recursive_exposure?(exposure)
```

Comment:
Where does this go exactly? It gets the first `nested_exposure` and sends it back through the method, at which point it hits the `else` path?

Reply:
`nested_exposures` can contain `nested_exposures`.  This recursively digs down through all of these until if finds the actual `nested_exposure`, then it hits the `else`.